foresight/environment/circleci/circleci_environment_info_provider.py

```diff
@@ -0,0 +1,56 @@
+from foresight.environment.git.git_helper import GitHelper
+from foresight.environment.environment_info import EnvironmentInfo
+from foresight.utils.test_runner_utils import TestRunnerUtils
+import os, logging
+from foresight.utils.generic_utils import print_debug_message_to_console
+
+LOGGER = logging.getLogger(__name__)
+
+class CircleCIEnvironmentInfoProvider:
+    ENVIRONMENT = "CircleCI"
+    CIRCLE_REPOSITORY_URL_ENV_VAR_NAME = "CIRCLE_REPOSITORY_URL"
+    CIRCLE_BRANCH_ENV_VAR_NAME = "CIRCLE_BRANCH"
+    CIRCLE_SHA1_ENV_VAR_NAME = "CIRCLE_SHA1"
+    CIRCLE_BUILD_URL_ENV_VAR_NAME = "CIRCLE_BUILD_URL"
+    CIRCLE_BUILD_NUM_ENV_VAR_NAME = "CIRCLE_BUILD_NUM"
+
+    @classmethod
+    def get_test_run_id(cls, repo_url, commit_hash):
+        configured_test_run_id = TestRunnerUtils.get_configured_test_run_id()
+        if configured_test_run_id:
+            return configured_test_run_id
+        build_url = os.getenv(cls.CIRCLE_BUILD_URL_ENV_VAR_NAME)
+        build_num = os.getenv(cls.CIRCLE_BUILD_NUM_ENV_VAR_NAME)
+        if build_url or build_url:
```

Comment:
should be **if build_url or build_num**:

Reply:
fixed.